ui/src/containers/BasicDetails/index.tsx

```diff
@@ -26,21 +26,34 @@ import { IconSize } from '@patternfly/react-icons';
 import { useObserver } from 'mobx-react';
 import { useParams } from 'react-router-dom';
 import { useMst } from '../../store/root';
-import { IResource } from '../../store/resource';
+import { IResource, IResourceStore } from '../../store/resource';
 import { ITag } from '../../store/tag';
 import { Icons } from '../../common/icons';
 import Icon from '../../components/Icon';
 import TooltipDisplay from '../../components/TooltipDisplay';
 import Rating from '../Rating';
 import { titleCase } from '../../common/titlecase';
 import './BasicDetails.css';
+import { assert } from '../../store/utils';
+import { ICatalogStore } from '../../store/catalog';
 
 const BasicDetails: React.FC = () => {
-  const { resources } = useMst();
-  const { catalog, kind, name } = useParams();
+  const { resources, catalogs }: { resources: IResourceStore; catalogs: ICatalogStore } = useMst();
+  const { catalog, kind, name }: { catalog: string; kind: string; name: string } = useParams();
 
   const resourceKey = `${catalog}/${titleCase(kind)}/${name}`;
-  const resource: IResource = resources.resources.get(resourceKey);
+  const resource: IResource | undefined = resources.resources.get(resourceKey);
+  assert(resource);
```

Comment:
Why are we asserting here ??

Reply:
since resource can be `undefined` as well

Reply:
I think since we are using `useObserver` function we don't need to handle that case